pkg/controller/daemon/util/daemonset_util.go

```diff
@@ -99,6 +99,11 @@ func CreatePodTemplate(template v1.PodTemplateSpec, generation *int64, hash stri
 		})
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ScheduleDaemonSetPods) {
+		// If scheduled by daemonset controler, set `.spec.schedulerName` to it.
+		newTemplate.Spec.SchedulerName = "daemonset-controller"
```

Comment:
shouldn't this go in the block where we set the node name if features.ScheduleDaemonSetPods is disabled?

are scheduler names namespaced (prefixed with `kubernetes.io/...`)? `default-scheduler` is a reserved value since that existed when the field was introduced, but it doesn't seem like we should suddenly claim another unnamespaced string

Reply:
> shouldn't this go in the block where we set the node name if features.ScheduleDaemonSetPods is disabled?

I'm thinking `.spec.SchedulerName` should be part of template, so I put it here :)

> are scheduler names namespaced (prefixed with kubernetes.io/...)? default-scheduler is a reserved value since that existed when the field was introduced, but it doesn't seem like we should suddenly claim another unnamespaced string

en... I think we should make it namespaced :)

Reply:
@msau42 does setting this to a value other than default-scheduler disrupt anything volume-related (attaching, binding, local volume stuff, etc)?

Reply:
The volume components do not look at the pod scheduler name.  

However, if the default scheduler does not process the pod (which is the current behavior when the feature is off), then none of the volume predicates will run (ie volume count, volume zone, volume binding).  And I think that is fine for Daemonsets because you can't really use those volume types in a Daemonset:

Volume count, zone: used by rwo volumes which doesn't work in a Daemonset because they can't be attached to more than one node at a time
Volume binding: used by topology aware PVCs which are currently local volumes and rwo zonal volumes.  Which don't work in a Daemonset because they all share the same PVC and are rwo.